### ISSUE TITLE
DrugExposureDiagnostics failed

### ISSUE THREAD

**Issue by GrigorisPapapostolou** [darwin-eu/CdmOnboarding#12]
When running the `cdmOnboarding` function with the following settings, an error occurs during the execution of `DrugExposureDiagnostics`, resulting in an empty section in the report for this diagnostic:

 ~~~
results <- CdmOnboarding::cdmOnboarding(
 connectionDetails = connectionDetails,
 cdmDatabaseSchema = 'omop',
 resultsDatabaseSchema = 'results',
 databaseId = '...',
 authors = '...',
 runVocabularyChecks = TRUE,
 runDataTablesChecks = TRUE,
 runPerformanceChecks = TRUE,
 runWebAPIChecks = TRUE,
 runDedChecks = TRUE,
 baseUrl = 'http://localhost:8080/WebAPI',
 outputFolder = outputFolder,
 dqdJsonPath = '...',
 smallCellCount = 5
)
~~~

Here is the **error** that appears in the logs:
 ~~~
...
2024-07-24 08:17:54	[Main thread]	INFO	CdmOnboarding	.execute	Running DED checks
2024-07-24 08:17:54	[Main thread]	INFO	CdmOnboarding	.runDedChecks	Starting execution of DrugExposureDiagnostics for 11 ingredients
2024-07-24 08:19:07	[Main thread]	ERROR	CdmOnboarding	3	Execution of DrugExposureDiagnostics failed: [1m[33mError[39m in `dplyr::collect()`:[22m [38;5;232m[33m![38;5;232m Failed to collect lazy table.[39m [1mCaused by error in `value[[3L]]()`:[22m [33m![39m org.postgresql.util.PSQLException: ERROR: syntax error at or near ","   Position: 11 
2024-07-24 08:19:07	[Main thread]	INFO	CdmOnboarding	.execute	Done.
2024-07-24 08:19:07	[Main thread]	INFO	CdmOnboarding	.execute	Complete CdmOnboarding took 13.26 minutes
...
 ~~~

**Environment**
* Ubuntu 22.04.4 LTS
* R version 4.2.3
* PostgreSQL 14.12
* CdmOnboarding version 3.2.0

**Comment by MaximMoinat** [comment]
@GrigorisPapapostolou Could you execute DrugExposureDiagnostics in isolation and share the full trackeback?
```R
library(DBI)
library(RPostgres)
library(DrugExposureDiagnostics)

server_parts <- strsplit(connectionDetails$server(), "/")[[1]]

connection <- DBI::dbConnect(
  RPostgres::Postgres(),
  dbname = server_parts[2],
  host = server_parts[1],
  user = connectionDetails$user(),
  password = connectionDetails$password()
)

cdm <- CDMConnector::cdm_from_con(
  connection,
  cdm_schema = cdmDatabaseSchema,
  write_schema = scratchDatabaseSchema
)

options(error = traceback)
DrugExposureDiagnostics::executeChecks(
  cdm = cdm,
  ingredients = c(
    528323,
    954688,
    968426,
    1119119,
    1125315,
    1139042,
    1140643,
    1154343,
    1550557,
    1703687,
    40225722
  ),
  checks = c("missing", "exposureDuration", "type", "route", "dose", "quantity", "diagnosticsSummary"),
  minCellCount = 5,
  sample = 1e+06,
  earliestStartDate = "2010-01-01"
)
```

**Comment by GrigorisPapapostolou** [comment]
I successfully executed the code mentioned earlier and have attached the output. Should I run the cdmOnboarding package again?
[output.txt](https://github.com/user-attachments/files/16410185/output.txt)

**Comment by MaximMoinat** [comment]
Thanks @GrigorisPapapostolou, that looks like a successful run. 
What version of DrugExposureDiagnostics did you use for this run? I suspect that this was using an older version of DED compared to what was executed as part of CdmOnboarding.

**Comment by GrigorisPapapostolou** [comment]
The versions of the packages DrugExposureDiagnostics and CdmOnboarding that I am using are '1.0.6' and '3.2.0', respectively.

**Comment by MaximMoinat** [comment]
I don't expect the version to be the issue, but could you try installing release [v1.0.7 of DED](https://github.com/darwin-eu/DrugExposureDiagnostics/releases/tag/v1.0.7) (released this week), and running CdmOnboarding again. To save time, you can disable the other checks:
```R
results <- CdmOnboarding::cdmOnboarding(
  connectionDetails = connectionDetails,
  cdmDatabaseSchema = cdmDatabaseSchema,
  resultsDatabaseSchema = resultsDatabaseSchema,
  oracleTempSchema = oracleTempSchema,
  databaseId = databaseId,
  authors = authors,
  smallCellCount = smallCellCount,
  baseUrl = baseUrl,
  outputFolder = outputFolder,
  runDedChecks = T,
  runVocabularyChecks = F,
  runDataTablesChecks = F,
  runPerformanceChecks = F,
  runWebAPIChecks = F,
  verboseMode = TRUE
)```

**Comment by IdISBa-PRISIB-Pau** [comment]
I had the same problem. In my case it was caused by observation_period where observation_period_start_date > observation_period_end_date but the error message from CdmOnboarding wouldn't say so. Executing DrugExposureDiagnostics in isolation as suggeted exposed the problem

**Comment by MaximMoinat** [comment]
This issue was solved with release 3.2.1, by disabling the validation done by CdmConnector
